### ISSUE TITLE
Question about argument span.

### ISSUE THREAD

**Issue by fairy-of-9** [pitrack/arglinking#4]
Hi!

I have some questions about code.
I'm interested in extracting spans out of the text.
I can't debug correctly because I can't execute the code, so I leave a question.

[this line](https://github.com/pitrack/arglinking/blob/5f4677efe580e2d22915d66be26ceff331a3b2c2/miso/models/arglinking.py#L230) is probably what I'm interested in.
I think, `arg_spans` in the line are all candidate argument text spans.

Here's what I'm curious about in this situation:
Q1:
in evaluation(inference) time,
Are `arg_spans` within-sentence spans up to a certain width (giving O(n) spans)?
In other words, Are `arg_spans` all candidate argument text spans?
Q2:
in train time,
Are `arg_spans` gold-argument spans or within-sentence spans up to a certain width (giving O(n) spans)?

Thanks!

**Comment by sethebner** [comment]
The answer to both your questions depends on the value of `use_gold_arguments` in the [config file](https://github.com/pitrack/arglinking/blob/5f4677efe580e2d22915d66be26ceff331a3b2c2/params/rams/gold_triggers%2Bgold_arguments/rams.yaml#L32), regardless of train time vs. inference time. If the flag is set to `True`, then we [use just the arguments annotated in the data](https://github.com/pitrack/arglinking/blob/5f4677efe580e2d22915d66be26ceff331a3b2c2/miso/data/dataset_readers/rams.py#L292). If the flag is set to `False`, then we [enumerate spans up to a certain width](https://github.com/pitrack/arglinking/blob/5f4677efe580e2d22915d66be26ceff331a3b2c2/miso/data/dataset_readers/rams.py#L296).

**Comment by fairy-of-9** [comment]
@sethebner 
Thanks for your reply!

Can you tell me what the value of `use_gold_arguments` is for the best performance of the model?
For example, 
train time : `True`, inference time : `False`

**Comment by sethebner** [comment]
Generally, performance will be better with `use_gold_arguments` set to `True` because it simplifies the task the model is learning: the model just has to link arguments with the correct role instead of also having to find the right argument spans. Nearly all the results in the paper use `use_gold_arguments = True` for training and inference, except for the bottom section of Table 2 ("distractor args" uses an expanded set of given candidate argument spans, and "no given args" uses enumeration).

**Comment by fairy-of-9** [comment]
Thank you so much for your kind reply.
Thank you for your research!
